docs/ldml/tr35-keyboards.md

```diff
@@ -1090,19 +1090,19 @@ This attribute may be escaped with `\u` notation, see [Escaping](#Escaping).
 ```xml
 <keyboard3>
     <keys>
-        <key id="a" to="a" longpress="\u{0301} \u{0300}" />
-        <key id="shift" switch="shift" />
+        <key id="a" output="a" longpress="\u{0301} \u{0300}" />
+        <key id="shift" layerId="shift" />
```

Comment:
Note: a limitation of longpress is we can't do e.g. `longpress="ctrl"` to get to the ctrl layer

Reply:
One way to solve this _if needed_ would be to make the `longpress` refer to key ids

Reply:
That would certainly be elegant structurally, but it would then require another lot of key elements to do the mapping to outputs, and _those_ keys would not be able to have longpress themselves, etc. Would require a significant reshuffle of the current design.

Reply:
They could. Longpress keys could be switching layers, and - if the options are presented in the UI - have multi tap, flicks or more longpress keys (nobody says the UI must dismiss as soon as you touch it) or assigned a `display` element. Implementations can simply not support it. I am not sure you need _that_ much structural redesign, but also I am not sure how useful this would be and the cost of that is more verbosity in the xml, hence emphasis on "if needed".

Compare this with multi tap situation. In this case, if you wanted to allow layer switching keys or long press keys, then you need significant implementation redesign as things change under your hands, and allowing "nested" multi tap does not make sense in the first place.

Reply:
My feeling on this is: layering gestures on gestures would lead to significant implementation complexity, a big decrease in usability and discoverability, and very little additional benefit.

As far as I can tell, only the `layerId` attribute is important as an attribute for any gesture outputs. But if we did refactor in this way, then it would be nice to also specify the `longPressDefault` as an attribute of the longPress rather than the pattern we have right now `<longPress default='true'>`.

@DavidLRowe noted that there are extant examples of keyboards in the Keyman keyboards repository that have layerId-type actions assigned to longpress. My quick and somewhat rough search located at least 81 keyboards. There are two kinds of uses:

1. Keyboards that have a modifier key with additional layer switch keys in the longpress, e.g. [koreguaje](https://github.com/keymanapp/keyboards/blob/59c44e8384b736f5f8e51a0af39f91bc677f0cce/experimental/k/koreguaje/source/koreguaje.keyman-touch-layout#L248-L269). As far as I have checked, these are mostly translations from desktop layouts.

2. Keyboards that emit an output and then swap to another layer. This pattern is widely used, e.g. in [naijatype](https://github.com/keymanapp/keyboards/blob/1729d48b84c535d0f783667ee76c0dc9c50259a5/experimental/n/naijatype/source/naijatype.keyman-touch-layout#L10-L78). This is an important usability pattern -- and restricting that ability to base keys seems like a poor outcome.

Given the current restriction we have with CLDR's XML structure, supporting this would lead to significantly more verbose files.

Is it too late to be shuffling these deck chairs?

Reply:
maybe for consistency, the flick output should _also_ be a key.

That would allow `<display>` to affect flick output. 

> Is it too late to be shuffling these deck chairs?

I think the too-late ship has already sailed in a sense, that we're already renaming elements.

I'll update.